### ISSUE TITLE
Adjust /etc/ssh/sshd_config automatically

### ISSUE THREAD

**Issue by toddr** [AlmaLinux/leapp-repository#12]
We noticed that if you do not have `PermitRootLogin` set in `/etc/ssh/sshd_config`, upgrades are blocked.

I suggest that instead, the file should be updated if the setting is missing to say: `PermitRootLogin yes`, which is the CentOS 7 default. Then you don't need to block on this.

**Comment by prilr** [comment]
In technical terms, this can be done without much difficulty.

That being said, I think that it should only be done when there was no `PermitRootLogin` present in the global config.
We wouldn't want to override a global `PermitRootLogin no` or `PermitRootLogin prohibit-password` automatically.

So in cases with an explicit `PermitRootLogin` we would preserve the current behavior, and with an implicit default, we would set a new value automatically and let the user know through a non-blocking report message.
@toddr Does that sound about right?

**Comment by toddr** [comment]
Correct. The blocker is when it is not, explicitly, set. My suggestion is that the blocker should never assert because if it is not set, then we should set it to the default for RHEL 7

**Comment by andrewlukoshko** [comment]
It seems like PermitRootLogin is not blocker anymore at least in leapp 0.14.0 / leapp-repository 0.16.0 when sshd_config is untouched.
Have you tried latest packages from http://repo.almalinux.org/elevate/el7/x86_64/ ?

**Comment by toddr** [comment]
Actually the change is now:

```
        if not config.permit_root_login and config.modified:
```

Which means the user won't be notified if permit_root_login is unmodified. This means that on upgrade, they will switch logic from `PermitRootLogin yes` to `PermitRootLogin no` and potentially get locked out. Isn't that bad?

**Comment by prilr** [comment]
That makes sense if we're expecting default config values provided to change with the upgrade - but I don't think that's something that can be relied upon.
